src/robot/parsing/populators.py

```diff
@@ -199,17 +199,28 @@ def _extension_is_accepted(self, ext, incl_extensions):
             return ext in incl_extensions
         return ext in READERS
 
-    def _is_included(self, path, base, ext, incl_extensions, incl_suites):
+    def _is_included(self, path, datadir, base, ext, incl_extensions, incl_suites):
         if base.startswith(self.ignored_prefixes):
             return False
         if os.path.isdir(path):
             return base not in self.ignored_dirs or ext
         if not self._extension_is_accepted(ext, incl_extensions):
             return False
-        return self._is_in_included_suites(base, incl_suites)
+        longname = self._get_longname(base, datadir, incl_suites)
+        return self._is_in_included_suites(base, incl_suites, longname)
 
-    def _is_in_included_suites(self, name, incl_suites):
-        return not incl_suites or incl_suites.match(self._split_prefix(name))
+    def _is_in_included_suites(self, name, incl_suites, longname=None):
+        return not incl_suites or incl_suites.match(longname or self._split_prefix(name), longname)
```

Comment:
What's the motivation to use `longname` in two places? If `longname` is given, this call is `incl_suites.match(longname, longname)` and that looks strange.

Reply:
Yes, you are right! Fixed it.